qipr_approver/approver/workflows/contact_person.py

```diff
@@ -0,0 +1,117 @@
+from itertools import chain
+
+from django.db.models import Q
+
+from approver import utils
+from approver.models import Person, Contact
+
+def get_collaborators_from_form(form, user):
+    return __process_tags_get_people(form, 'collaborator', user)
+
+def get_advisors_from_form(form, user):
+    return __process_tags_get_people(form, 'advisor', user)
+
+def __process_tags_get_people(form, prop, user):
+    """
+    Using the form data tags we get back this function gets the right
+    person object
+
+    Possibilities:
+    ---------------
+    1. tag is never before seen email => make new contact and person
+    2. tag is contact guid => get matching contact and make new person from that
+    3. tag is person guid => get matching person
+
+    return the right people
+    """
+    tags = __process_tags(form, prop)
+
+    new_contacts = [tag for tag in tags if __is_email(tag)]
+    new_contact_people = __handle_email_tags(new_contacts, user)
+
+    # dealt with new contacts
+    tags = [tag for tag in tags if not (tag in new_contacts)]
+
+    # now we have guid tags only
+    filter_object = __get_filter_object(tags)
+
+    contact_matches = [model for model in Contact.objects.filter(filter_object).select_related('person')]
+    person_matches = [model for model in Person.objects.filter(filter_object)]
+
+    people_from_contacts = __handle_contact_matches(contact_matches, user)
```

Comment:
nice, so let me check my understanding:
If email, it's new to the system so go make a new contact and corresponding person.
if guid:
A guid is unique, so we check both the contacts and person table for it.

if it is a contact, we grab the person associated with them. If it is a person, we are good.
We then jam the 3 potential people lists together and return it.